### PR TITLE
Added checkbox secure to callback option dialog. 

### PR DESCRIPTION
**Pull request by denniskniep**
This allows the usage of https when the callbackadress is requested with the method getCallbackAddress(). 

Only https in Ui and for creating the callback adress. Need additionally PR #3975 to work as expected.

### CODE REVIEW

File: src/org/zaproxy/zap/extension/callback/ExtensionCallback.java [hunk 1]
```diff
@@ -98,12 +98,14 @@ private void restartServer(int port) {
 
     public String getCallbackAddress() {
         String addr = this.getCallbackParam().getRemoteAddress();
+        boolean isSecure = this.getCallbackParam().isSecure();
+        String schema = isSecure ? "https" : "http";
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/callback/ExtensionCallback.java]: `scheme`.

File: src/org/zaproxy/zap/extension/callback/ExtensionCallback.java [hunk 2]
```diff
@@ -98,12 +98,14 @@ private void restartServer(int port) {
...
         if (addr.contains(":")) {
             // Looks like its IPv6
-            return "http://[" + addr + "]:" + actualPort + "/";
+            return schema + "://[" + addr + "]:" + actualPort + "/";
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/callback/ExtensionCallback.java]: Worth remove the code duplication: `addr = "[" + addr + "]";` (and drop the IPv4 comment).

Comment (denniskniep) [reply src/org/zaproxy/zap/extension/callback/ExtensionCallback.java]: You mean this way?

```     
boolean ipv6 = addr.contains(":");
String ipAddr = ipv6 ? "[" + addr + "]" : addr;   
return schema + "://" + ipAddr + ":" + actualPort + "/";
```

Comment (thc202) [reply src/org/zaproxy/zap/extension/callback/ExtensionCallback.java]: Was suggesting:
```Java
if (addr.contains(":")) {
    // Looks like its IPv6
    addr = "[" + addr + "]";
}
return scheme + "://" + addr + ":" + actualPort + "/";
```
but that's fine as well (although the variable `ipAddr` should have other name, `addr` might be a hostname).

**Comment by thc202** [comment]
Thanks!

**Pull request merged**